### DISCUSSION TITLE
Why baseUrl of SessionProvider not take effect?

### DISCUSSION THREAD

**Discussion by Jokcy** [nextauthjs/next-auth#6385] [Help]
https://github.com/nextauthjs/next-auth/blob/8f7145801a8963d183b76cb49d0a95c5a83548a7/packages/next-auth/src/react/types.ts#L67

There is a `baseUrl` prop for SessionProvider but it did not being used actually. I suppose it should work just like `basePath` prop

https://github.com/nextauthjs/next-auth/blob/8f7145801a8963d183b76cb49d0a95c5a83548a7/packages/next-auth/src/react/index.tsx#L334

And reading env variables without `NEXT_APP_` prefix in client side is not possibe, so I guess code here is useless?

https://github.com/nextauthjs/next-auth/blob/8f7145801a8963d183b76cb49d0a95c5a83548a7/packages/next-auth/src/react/index.tsx#L50

**Comment by Jokcy** [comment]
Also here:
https://github.com/nextauthjs/next-auth/blob/8f7145801a8963d183b76cb49d0a95c5a83548a7/packages/next-auth/src/client/_utils.ts#L53

Should return `${__NEXTAUTH.baseUrl}${__NEXTAUTH.basePath}`, is there any problem we support different domain (same root domain and put session cookie under root domain)?

**Comment by Jokcy** [comment]
Please give some feedback, I can PR for this if there is no other issue to support baseUrl.

**Comment by ThangHuuVu** [comment]
Thanks for raising this, it seems to me that it is not a common use case based on the community feedback. We shall revisit once we got enough interest 🙏
